cm.go

```diff
@@ -0,0 +1,70 @@
+package sachet
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// CMConfig is the configuration struct for CM provider
+type CMConfig struct {
+	ProductToken string `yaml:"producttoken"`
+}
+
+// CM contains the necessary values for the CM provider
+type CM struct {
+	CMConfig
+}
+
+var cmHTTPClient = &http.Client{Timeout: time.Second * 20}
+
+// NewCM creates and returns a new CM struct
+func NewCM(config CMConfig) *CM {
+	return &CM{config}
+}
+
+// Send sends SMS to n number of people using Bulk SMS API
+func (c *CM) Send(message Message) error {
+	smsURL := "https://gw.cmtelecom.com/v1.0/message"
+
+	var dataMap map[string]interface{}
```

Comment:
structured types  are safer than mant map[string]interface{}. Something like this anonymous one:

```go
	payload := struct {
		Authentication string `json:"authentication"`
		Number string
		From string
	}{
		Authentication: "secret",
		Number: "32939854",
	}
	// etc
	data, err := json.Marshal(payload)

```